pkg/controller/daemon/util/daemonset_util.go

```diff
@@ -99,6 +99,11 @@ func CreatePodTemplate(template v1.PodTemplateSpec, generation *int64, hash stri
 		})
 	}
 
+	if !utilfeature.DefaultFeatureGate.Enabled(features.ScheduleDaemonSetPods) {
```

Comment:
I think it makes more sense to do this here, where we already have the logic to delegate to the default scheduler or not:

https://github.com/kubernetes/kubernetes/blob/16c5ba4114bf6443c2ce8c0104ad520dbcece6ce/pkg/controller/daemon/daemon_controller.go#L993-L1007

Reply:
done